### PR TITLE
[Feature] Support spark load without broker

### LINKED ISSUE #9229: [Feature] Support spark load without broker

### PR DESCRIPTION
**Pull request by xiaoyong-z**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9229

## Problem Summary(Required) ：

Support spark load without broker

### CODE REVIEW

File: be/src/fs/fs.cpp [hunk 1]
```diff
@@ -92,15 +93,15 @@ StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::st
     return Status::NotSupported(fmt::format("No FileSystem associated with {}", uri));
 }
 
-StatusOr<std::shared_ptr<FileSystem>> FileSystem::CreateSharedFromString(std::string_view uri) {
+StatusOr<std::shared_ptr<FileSystem>> FileSystem::CreateSharedFromString(std::string_view uri, FSOptions options) {
```
Comment (chaoyli) [inline be/src/fs/fs.cpp:96]: const FSOptions&

File: be/src/fs/fs.cpp [hunk 2]
```diff
@@ -74,15 +75,15 @@ inline bool is_staros_uri(std::string_view uri) {
     return false;
 }
 
-StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::string_view uri) {
+StatusOr<std::unique_ptr<FileSystem>> FileSystem::CreateUniqueFromString(std::string_view uri, FSOptions options) {
```
Comment (chaoyli) [inline be/src/fs/fs.cpp:78]: const FSOptions&

File: be/src/fs/fs_hdfs.cpp [hunk 3]
```diff
@@ -205,20 +372,25 @@ StatusOr<std::unique_ptr<RandomAccessFile>> HdfsFileSystem::new_random_access_fi
...
     if (file == nullptr) {
         return Status::InternalError(fmt::format("hdfsOpenFile failed, file={}", path));
     }
     auto stream = std::make_shared<HdfsInputStream>(handle.hdfs_fs, file, path);
     return std::make_unique<RandomAccessFile>(std::move(stream), path);
 }
 
-std::unique_ptr<FileSystem> new_fs_hdfs() {
-    return std::make_unique<HdfsFileSystem>();
+std::unique_ptr<FileSystem> new_fs_hdfs(FSOptions& options) {
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:392]: const FSOptions&

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 62 (66.13%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[315]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|6|00.00%|[195, 201, 202, 214, 225, 226]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[315]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|6|00.00%|[195, 201, 202, 214, 225, 226]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/fs/HdfsUtil.java [hunk 4]
```diff
@@ -211,18 +211,19 @@ public static void writeFile(String srcFilePath, String destFilePath,
...
             }
         } catch (IOException e) {
-            String failMsg = "Read file exception. filePath=" + srcFilePath;
+            String failMsg = "Write file exception. srcPath = " + srcFilePath + 
+                    ", destPath = " + destFilePath;
             LOG.warn(failMsg, e);
             throw new UserException(failMsg);
         } finally {
-            // close broker file writer and local file input stream
+            // close broker file writer and local file write stream
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/fs/HdfsUtil.java:219]: is this comment right?

Comment (xiaoyong-z) [reply fe/fe-core/src/main/java/com/starrocks/fs/HdfsUtil.java:219]: a mistake here, should be input

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[315]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|6|00.00%|[195, 201, 202, 214, 225, 226]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Review by chaoyli — APPROVED** [review]

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[315]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|6|00.00%|[195, 201, 202, 214, 225, 226]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[315]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|6|00.00%|[195, 201, 202, 214, 225, 226]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|13|21|61.90%|[190, 520, 521, 522, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkEtlJobHandler.java|8|12|66.67%|[128, 267, 330, 360]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkRepository.java|10|15|66.67%|[165, 181, 232, 300, 314]|
|:large_blue_circle:|com/starrocks/catalog/SparkResource.java|10|14|71.43%|[194, 195, 197, 319]|

**Pull request merged**